### ISSUE TITLE
Issue creating flow outputs with LaharZ

### ISSUE THREAD

**Issue by ailsa-naismith** [Keith1815/LaharZ#12]
I have had an issue in creating flow outputs from LaharZ. I have installed LaharZ from Conda on a MacOS, initiated LaharZ, and run initiation points without trouble. I then run "Create Flow Outputs" and follow instructions for inputs required. On pressing "Create Flow Outputs", LaharZ freezes and the command line reveals the following error:

`Exception in Tkinter callback`
`Traceback (most recent call last):`
  `File "/opt/miniconda3/envs/laharz-test-00/lib/python3.12/tkinter/__init__.py", line 1968, in __call__
    return self.func(*args)`
           `^^^^^^^^^^^^^^^^`
 `File "/opt/miniconda3/envs/laharz-test-00/lib/python3.12/site-packages/laharz/laharz.py", line 2196, in press_lahar_button
    error = gen_lahars() `
            `^^^^^^^^^^^^`
  `File "/opt/miniconda3/envs/laharz-test-00/lib/python3.12/site-packages/laharz/laharz.py", line 2760, in gen_lahars
    log_msg('Parameter: pc1_value; C1 Parameter; Value: ' + self.pc1_value, screen_op = False)`
            `~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~`
`TypeError: can only concatenate str (not "float") to str>`

I think this is an operational error between two different types: the program tries to concatenate a string with a floating point number, but only strings can be concatenated with strings. The '^' above suggests error is in concatenating 'self.pc1_value', and that the solution would be to convert 'self.pc1_value' from float to string. However, I haven't worked out how to do this in Conda - or confirmed if this is indeed the reason why the "Create Flow Outputs" is not completing! Any insights appreciated; I imagine this is a pretty basic issue.

**Comment by Keith1815** [comment]
Hi Ailsa, nice to hear from you and thanks for letting me know. Delighted you are using this. If you need any help reach out.

I have been able to reproduce the error. It is a simple fix. I will try to get it done today or tomorrow and will let you know. 

Cheers, Keith

**Comment by Keith1815** [comment]
Hi Aisla, I think you have this error when you use custom values for the coefficients. To work around this, you can create your own scenario in Maintain System Parameters (python -m laharz maintain) with it's own coefficients. The details are in the documentation (p33). In the meantime, I will try to correct it. Cheers, Keith

**Comment by Keith1815** [comment]
Hi Ailsa, this is now fixed. A little more quickly than I expected. If you reinstall Laharz (conda install -c keith1815 laharz) it should be fine. Let me know how you get on. 

Cheers, Keith

**Comment by ailsa-naismith** [comment]
Hello Keith, thank you very much for your quick reply and for taking the time to fix. Really enjoying using the program. It's fixed! LaharZ is creating flow outputs now. 
I followed your guidance by reinstalling LaharZ, and then defining custom values for coefficients c1 and c2 in Maintain System Parameters. I'll redefine these for future scenarios. 

I experienced a brief hiccup with the following error that ends:
` File "/opt/miniconda3/envs/laharz-test-00/lib/python3.12/site-packages/geopandas/io/file.py", line 173, in _is_zip
    parsed = fiona.path.ParsedPath.from_uri(path)`
             `^^^^^^^^^^`
`AttributeError: module 'fiona' has no attribute 'path'. Did you mean: 'Path'?`
Which happens if you don't have the latest version of the geopandas package installed. I fixed the issue using [this](https://stackoverflow.com/questions/78949093/how-to-resolve-attributeerror-module-fiona-has-no-attribute-path) StackOverflow thread.

Will be in touch in the future!

**Comment by Keith1815** [comment]
Hi, that's brilliant. Glad to hear its working and you enjoy using it. Would love to see what you're using it for if you have time. 

On your points:
1) on the c1, c2 parameters you should be able to either:
- use the standard scenarios (Lahar, Pyroclastic, Debris) and the associated parameters; or
- use the custom option to enter any parameters you like; or
- create your own scenarios with your own coefficient values in the maintain system parameters screen

2) On Geopandas/Fiona.
- On my version I have Geopandas 1.0.1 installed and I don't get that error.
- The package requirements are for version >= 0.14.0
- I guess the interdependencies between these packages is a bit of a moving feast. I'll update the dependency to version 1.0.1 for future releases, just in case there is problem with older versions of Geopandas.

Thanks again for the feedback. Cheers, Keith

**Comment by ailsa-naismith** [comment]
Hello, happy to do so - will follow up with an email detailing why we're using it!
1. On the c1, c2 parameters I've successfully created my own scenarios with "Maintain System Parameters" screen. I'll work through the software again to see if I succeed following previous bullets. 
2. Correct, you stated in documentation (pg 8) the required packages and their version numbers (inc. geopandas =0.14.3) and that if users haven't got these versions, they may have errors. I probably installed an earlier version of geopandas, and ran LaharZ without checking. Again, page 8 has useful advice for checking version numbers of packages and for reinstalling packages to avoid/correct these simple errors.
Many thanks!
